src/stash.c

```diff
@@ -159,7 +164,37 @@ struct stash_update_rules {
 	bool include_ignored;
 };
 
+/* similar to git_index_add_bypath but able to operate on any
+ * index without making assumptions about the repository's index
+ */
+static int stash_to_index(
+	git_repository *repo,
+	git_index *index,
+	const char *path)
+{
+	git_index *repo_index;
+	git_index_entry entry = {{0}};
+	struct stat st;
+	int error;
+
+	if (!git_repository_is_bare(repo) &&
+		(error = git_repository_index__weakptr(&repo_index, repo)) < 0)
```

Comment:
I'd like to finally settle on a common style when it comes to multi-line conditionals. We currently have two different styles in our code base. The first one aligned with a tab

```
if (!git_repository_is_bare(repo) &&
	(error = git_repository_index__weakptr(&repo_index, repo)) < 0)
	return error;
```

and the same thing aligned with four spaces instead

```
if (!git_repository_is_bare(repo) &&
    (error = git_repository_index__weakptr(&repo_index, repo)) < 0)
	return error;
```

I'd say the second one is superior, as you can clearly distinguish conditionals from the body. Especially when there are multiple more than two lines in a conditional, this becomes rather valuable and makes code a lot easier to read, in my opinion.

Reply:
> I'd like to finally settle on a common style when it comes to multi-line conditionals.  We currently have two different styles in our code base

I... guess?  I agree that we _shouldn't_ have two different styles, and yes, some things have slipped in here because _it really doesn't matter that much_ (more on that below).  But it feels like this was a solved decision.  We took git's style, which, from a cursory grep, is _typically_ indenting these with tabs.  And we _typically_ indent with tabs.

It seems like anything that's slipped in has done just that - slipped in.

> I'd say the second one is superior, as you can clearly distinguish conditionals from the body. Especially when there are multiple more than two lines in a conditional, this becomes rather valuable and makes code a lot easier to read, in my opinion.

I agree, but that's why I have my tabs set to four spaces.

But for the broader point:  it's _incredibly_ frustrating for contributors when they have to wait several weeks for a code review and then have that code review consist primarily of complaints about whitespace.  (This one was open three months ago!  _Three months_ is a long time, and yes, I realize that the complaints are not _only_ about whitespace.)

Does style matter?  Sure.  But they don't matter _nearly_ enough to stop the velocity of the project, or to turn people off from contributing.

Can you imagine waiting weeks to get a code review then it's picking nits at whitespace changes?   This is not the sign of a healthy community, and that person will probably never contribute again.

We need to fix this when it comes to reviewing pull requests.  This is not a complaint at you, individually; I do mean _we_, the project.

As for the whitespace issues: if you _really_ care about this, then add an automated step to the CI builds and give people something easy that they can run locally.  Otherwise, we need to relax, IMHO.

Reply:
>> I'd like to finally settle on a common style when it comes to multi-line conditionals. We currently have two different styles in our code base
> I... guess? I agree that we shouldn't have two different styles, and yes, some things have slipped in here because it really doesn't matter that much (more on that below). But it feels like this was a solved decision. We took git's style, which, from a cursory grep, is typically indenting these with tabs. And we typically indent with tabs.

It seems like anything that's slipped in has done just that - slipped in.

I might not have brought my intention along crlearly enough, but this was not a request for change but instead a clarification for what our agreed-upon style is. I had asked that specific question about how to indent multiline conditionals in the past and never got an answer (or at least I couldn't recall). So while I feel like the latter option is superior, I don't care which of both styles we use as long as it is consistent.

> I agree, but that's why I have my tabs set to four spaces.

Weird. I always thought the projects recommendation was eight spaces, and that's why I always had a specific setting for libgit2 to display as eight spaces. But in fact, we do recommend to set it to four spaces.

> But for the broader point: it's incredibly frustrating for contributors when they have to wait several weeks for a code review and then have that code review consist primarily of complaints about whitespace. (This one was open three months ago! Three months is a long time, and yes, I realize that the complaints are not only about whitespace.)

I know. And that's why I said in the past that I only add style-related critic in case there are other complaints about a PR.

> Does style matter? Sure. But they don't matter nearly enough to stop the velocity of the project, or to turn people off from contributing.
>
> Can you imagine waiting weeks to get a code review then it's picking nits at whitespace changes? This is not the sign of a healthy community, and that person will probably never contribute again.
> We need to fix this when it comes to reviewing pull requests. This is not a complaint at you, individually; I do mean we, the project.
>
> As for the whitespace issues: if you really care about this, then add an automated step to the CI builds and give people something easy that they can run locally. Otherwise, we need to relax, IMHO.

I'm definitely in favor of having an automated job to do this. While I agree that style shouldn't be the main concern, I think it in fact _is_ important for a project. I've seen enough projects where I wanted to contribute to, but after taking a quick peek into the implementation immediately bailed out as the code was so inconsistently formatted that I had problems to see what was going on. So it in fact is a double-edged sword: you need to care enough about style to not let the code base deteriorate and get hard to read, but you shouldn't be as strict to drive away contributors.

As you mentioned, I think the best way forward would be to have an automated job that does this immediately after a PR was submitted. If properly implemented, this would help both roles: the contributor can quickly fix up stuff while he is still familiar with the topic, and the maintainer can focus on the important parts.